packages/console/src/components/DateTime/index.tsx

```diff
@@ -10,21 +10,74 @@ const parseDate = (date: Nullable<string | number | Date>) => {
   return isValid(parsed) ? parsed : undefined;
 };
 
+type DateTimeFormatOptions = Intl.DateTimeFormatOptions;
+
 type Props = {
   readonly children: Nullable<string | number | Date>;
+  readonly format?: Partial<DateTimeFormatOptions>;
+};
+
+const defaultDateFormat: DateTimeFormatOptions = {
+  year: 'numeric',
+  month: 'numeric',
+  day: 'numeric',
+};
+
+const defaultDateTimeFormat: DateTimeFormatOptions = {
+  year: 'numeric',
+  month: 'numeric',
+  day: 'numeric',
+  hour: 'numeric',
+  minute: 'numeric',
+  second: 'numeric',
+  hour12: true,
 };
 
 /**
  * Safely display a date in the user's locale. If the date is invalid, it will display a dash.
+ * @param format - Optional Intl.DateTimeFormatOptions to customize the date format
  */
-export function LocaleDate({ children }: Props) {
-  return <span>{parseDate(children)?.toLocaleDateString() ?? '-'}</span>;
+export function LocaleDate({ children, format = {} }: Props) {
+  const date = parseDate(children);
+
+  if (!date) {
+    return <span>-</span>;
+  }
+
+  return (
+    <span>
+      {new Intl.DateTimeFormat(undefined, { ...defaultDateFormat, ...format }).format(date)}
```

Comment:
nit: Why don't we use `date-fns` format?

Reply:
Good, updated.